### PR TITLE
359 choreui limit chat message input length

### PR DESCRIPTION
**Pull request by andrewrisse**
Refactors LFTextArea component to limit input length and show errors when the user attempts to type past the input length.

I used the Carbon Components Svelte TextArea component and modified it to add our unique features (grow on new line, limit growth to maxRows then start scrolling, show errors on length limit).

Length limit is configurable via a environment variable.

Several of the changes are prettier changes from running npm run format.

### CODE REVIEW

**Comment by andrewrisse** [comment]
[max_input.webm](https://github.com/defenseunicorns/leapfrogai/assets/52644157/d2ad0801-7244-436c-bd90-c3f21b90aada)

File: src/leapfrogai_ui/src/lib/components/Message.test.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { render, screen } from '@testing-library/svelte';
-import { afterAll, afterEach, vi } from 'vitest';
+import { afterAll, afterEach, type MockInstance, vi } from 'vitest';
```
Comment (gerred) [inline src/leapfrogai_ui/src/lib/components/Message.test.ts:2]: TIL about this destructuring for type imports

**Review by gerred — APPROVED** [review]

**Comment by gerred** [comment]
@andrewrisse lgtm, just needs a merge conflict fix

**Review by gerred — APPROVED** [review]

File: packages/ui/zarf.yaml [hunk 2]
```diff
@@ -57,7 +57,11 @@ variables:
     default: 'true'
     prompt: true
     sensitive: false
-
+  - name: MESSAGE_LENGTH_LIMIT
+    description: The maximum length of a message to allow from the user to the backend
+    default: '10000'
```
Comment (BillyFigueroa) [inline packages/ui/zarf.yaml:62]: Hey Andrew, is there a benefit to using a string here vs a number?

Comment (andrewrisse) [reply packages/ui/zarf.yaml:62]: It's because when you have it in your .env file, it's parsed as a string when you import it

**Pull request merged**